error.go

```diff
@@ -156,6 +157,30 @@ func (e *Error) Cause() error {
 	return e.cause
 }
 
+// Is implements the helper interface for errors.Is
+// It first checks if target is an *Error and uses IsOfType to check if e Is target.
+// If target is not an *Error, it instead iterates through the error chain until it reaches a non-*Error error (or nil)
+// and calls errors.Is on that error and target.
+func (e *Error) Is(target error) bool {
+	// Check if target is an *Error, if so we can use IsOfType.
+	t := Cast(target)
+	if t != nil {
+		return e.IsOfType(t.Type())
+	}
+
+	// target is not an *Error, so iterate through the error chain until nil or a non-*Error error is reached and use
+	// errors.Is to check against that.
+	cause := e
+	for cause != nil {
+		next := Cast(cause.Cause())
+		if next == nil && cause.Cause() != nil {
+			return errors.Is(cause.Cause(), target)
```

Comment:
Note, I noticed that your CI is running tests against Go  `1.8` which does not have the `errors` package, but the CI appears to be broken.  Do ya'll need to support versions below `1.13` where this package was added? If so, I'll probably need to implement the same logic as `errors.Is` in this package rather than using the built-in one.

Reply:
@PeterIvanov BTW we currently have CI failing for master since we pull latest testify and it requires go 1.13 now. If we want to support go 1.8 we should not use modules to specify dependency version either... How about to require 1.13 as a minimum?
@pacejackson If we want to support earlier go versions (makes sense for project with a replacement, at least partial,  in newer versions of stdlib) there is a way - #29

Reply:
Yes, I only just noticed it while reviewing this PR. 
I suppose that with this change, we could make a major release version which requires `go 1.13`. Those with older go version will still be able to use older release. Does it seem reasonable to everyone?

Reply:
See comment below. I will review the code change itself in a short while.

Reply:
Shouldn't we alias `Cause` to `Unwrap` ? And then `Is` method will be just first `if` statement from this code.

Reply:
This was already discussed in the earlier version of this PR, see thread above.

Reply:
Yep, I've missed. But I still think correct `Unwrap` + simple `Is` method is more prefferable to overused single `Is`. Perhaps `Unwrap` should not be raw alias to `Cause`, but test transparency.

Reply:
`Is` seems to be the a better abstraction in my opinion, and is designed specifically for custom wrappers with its own logic, which suits errorx perfectly. 
On the other hand, I do not object strongly against moving the major part of this logic to `Unwrap()`. Let the PR author decide on this, provided the test coverage is good and transparency rules are respected in all cases.

Reply:
I think that sounds reasonable, to be clear, what ya'll are suggesting is go back to defining `Unwrap` rather than `Is` but have check the transparency and return the first non-transparent error?

Reply:
I can add little to my last comment. Personally, I would leave only `Is` method, changing its behaviour as suggested in review comments below. But, as long at it behaves the same, the choice is yours.

Reply:
Looks like, to be the same, Error.Unwrap should return nil if it is non-transparent.